### PR TITLE
Pd 3414/update deps

### PR DESCRIPTION
**Pull request by maneki-admin**
- Update package dependencies
- Run tests on PR, ecr push/deploy when push to master only
- Add dependabot config

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    schedule:
+      interval: "monthly"
+  - package-ecosystem: "docker"
+    directory: "/"
+    schedule:
+      interval: "monthly"
+  - package-ecosystem: "npm"
+    directory: "/"
+    schedule:
+      interval: "monthly"
```
Comment (tainguyentt) [inline .github/dependabot.yml:14]: Should we use weekly or monthly interval?

Comment (maneki-admin) [reply .github/dependabot.yml:14]: I dont have a strong opinion about this one. But in my experience, big number of repos * weekly schedule = hard to review + big number of github action minutes run.

File: packages/constants/package.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "lint": "eslint --fix 'src/**/*.ts'"
   },
   "peerDependencies": {
-    "@babel/runtime": "7.x"
+    "@babel/runtime": "^7.24.1"
```
Comment (chaitanyapotti) [inline packages/constants/package.json:21]: keep peer dependencies as 7.x

File: .github/dependabot.yml (hunk 1 above)
Comment (chaitanyapotti) [reply .github/dependabot.yml:14]: monthly is fine

**Review by tainguyentt — APPROVED** [review]
lgtm

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**